### ISSUE TITLE
ggml_new_object: not enough space in the context's memory pool

### ISSUE THREAD

**Issue by bobqianic** [ggerganov/whisper.cpp#1272]
```
C:\Users\qianp\Downloads\whisper.cpp_build-fix\bin\Release>C:\Users\qianp\Downloads\whisper.cpp_build-fix\bin\Release\bench.exe -m C:\Users\qianp\Downloads\whisper.cpp_build-fix\bin\Release\ggml-model-whisper-base.bin -w 2
ggml_new_object: not enough space in the context's memory pool (needed 301991760, available 268437504)
```

I just ran some tests and noticed that `bench.cpp` is corrupted. It seems this problem started with the commit [59a3d0c](https://github.com/ggerganov/whisper.cpp/commit/59a3d0cb576db605f76f82f07350647837e15c7a) on September 5th. I'm guessing the `ggml` updates weren't properly reflected in ~~`bench.cpp`~~ `whisper.cpp`.

**Comment by dereklll** [comment]
same problem on mac os.

**Comment by dereklll** [comment]
./talk-llama -mw ~/work/whisper.cpp/ggml-model-whisper-base.bin -ml ~/work/7B/chinese-alpaca-2-7b-16k-hf/ggml-model-f16.bin -p "You" -t 12 -l zh -c 0
whisper_init_from_file_no_state: loading model from '/Users/lll/work/whisper.cpp/ggml-model-whisper-base.bin'
whisper_model_load: loading model
whisper_model_load: n_vocab       = 51865
whisper_model_load: n_audio_ctx   = 1500
whisper_model_load: n_audio_state = 512
whisper_model_load: n_audio_head  = 8
whisper_model_load: n_audio_layer = 6
whisper_model_load: n_text_ctx    = 448
whisper_model_load: n_text_state  = 512
whisper_model_load: n_text_head   = 8
whisper_model_load: n_text_layer  = 6
whisper_model_load: n_mels        = 80
whisper_model_load: ftype         = 1
whisper_model_load: qntvr         = 0
whisper_model_load: type          = 2
whisper_model_load: mem required  =  310.00 MB (+    6.00 MB per decoder)
whisper_model_load: adding 1608 extra tokens
whisper_model_load: model ctx     =  140.66 MB
whisper_model_load: model size    =  140.54 MB
whisper_init_state: kv self size  =    5.25 MB
whisper_init_state: kv cross size =   17.58 MB
llama.cpp: loading model from /Users/lll/work/7B/chinese-alpaca-2-7b-16k-hf/ggml-model-f16.bin
llama_model_load_internal: format     = ggjt v1 (pre #1405)
llama_model_load_internal: n_vocab    = 55296
llama_model_load_internal: n_ctx      = 2048
llama_model_load_internal: n_embd     = 4096
llama_model_load_internal: n_mult     = 256
llama_model_load_internal: n_head     = 32
llama_model_load_internal: n_layer    = 32
llama_model_load_internal: n_rot      = 128
llama_model_load_internal: ftype      = 1 (mostly F16)
llama_model_load_internal: n_ff       = 11008
llama_model_load_internal: n_parts    = 1
llama_model_load_internal: model size = 7B
llama_model_load_internal: ggml ctx size =    0.09 MB
llama_model_load_internal: mem required  = 15009.11 MB (+ 1026.00 MB per state)
.
llama_init_from_file: kv self size  = 1024.00 MB

main: processing, 12 threads, lang = zh, task = transcribe, timestamps = 0 ...

init: found 1 capture devices:
init:    - Capture device #0: 'Logitech USB Headset H340'
init: attempt to open capture device 0 : 'Logitech USB Headset H340' ...
init: obtained spec for input device (SDL Id = 2):
init:     - sample rate:       16000
init:     - format:            33056 (required: 33056)
init:     - channels:          1 (required: 1)
init:     - samples per frame: 1024

main : initializing - please wait ...
ggml_new_object: not enough space in the context's memory pool (needed 910471808, available 805306368)
zsh: trace trap  ./talk-llama -mw ~/work/whisper.cpp/ggml-model-whisper-base.bin -ml  -p "You"

**Comment by dereklll** [comment]
./extra/bench-all.sh 
Usage: ./bench.sh [n_threads] [encoder-only]

Running memcpy benchmark

memcpy: 51.54 GB/s (1 thread)
sum:    -536871564.000000

Running ggml_mul_mat benchmark with 4 threads

  64 x   64: Q4_0     9.4 GFLOPS (128 runs) | Q4_1     9.8 GFLOPS (128 runs)
  64 x   64: Q5_0     9.9 GFLOPS (128 runs) | Q5_1     9.9 GFLOPS (128 runs) | Q8_0    13.9 GFLOPS (128 runs)
  64 x   64: F16     16.2 GFLOPS (128 runs) | F32     16.4 GFLOPS (128 runs)
 128 x  128: Q4_0   101.3 GFLOPS (128 runs) | Q4_1    96.3 GFLOPS (128 runs)
 128 x  128: Q5_0   100.8 GFLOPS (128 runs) | Q5_1   100.5 GFLOPS (128 runs) | Q8_0   105.9 GFLOPS (128 runs)
 128 x  128: F16    109.9 GFLOPS (128 runs) | F32    115.1 GFLOPS (128 runs)
 256 x  256: Q4_0   445.4 GFLOPS (128 runs) | Q4_1   383.3 GFLOPS (128 runs)
 256 x  256: Q5_0   438.6 GFLOPS (128 runs) | Q5_1   441.8 GFLOPS (128 runs) | Q8_0   456.5 GFLOPS (128 runs)
 256 x  256: F16    479.0 GFLOPS (128 runs) | F32    503.7 GFLOPS (128 runs)
 512 x  512: Q4_0   991.3 GFLOPS (128 runs) | Q4_1   925.4 GFLOPS (128 runs)
 512 x  512: Q5_0  1085.8 GFLOPS (128 runs) | Q5_1  1105.5 GFLOPS (128 runs) | Q8_0  1009.9 GFLOPS (128 runs)
 512 x  512: F16   1195.2 GFLOPS (128 runs) | F32   1507.0 GFLOPS (128 runs)
1024 x 1024: Q4_0  1823.7 GFLOPS (128 runs) | Q4_1  1671.1 GFLOPS (128 runs)
1024 x 1024: Q5_0  1924.8 GFLOPS (128 runs) | Q5_1  2086.6 GFLOPS (128 runs) | Q8_0  1814.1 GFLOPS (128 runs)
1024 x 1024: F16   2129.2 GFLOPS (128 runs) | F32   2335.1 GFLOPS (128 runs)
2048 x 2048: Q4_0  3255.9 GFLOPS (128 runs) | Q4_1  2936.0 GFLOPS (128 runs)
2048 x 2048: Q5_0  3221.4 GFLOPS (128 runs) | Q5_1  3284.2 GFLOPS (128 runs) | Q8_0  3414.5 GFLOPS (128 runs)
2048 x 2048: F16   3432.8 GFLOPS (128 runs) | F32   3535.2 GFLOPS (128 runs)
4096 x 4096: Q4_0  3369.9 GFLOPS ( 25 runs) | Q4_1  3164.7 GFLOPS ( 24 runs)
4096 x 4096: Q5_0  3296.3 GFLOPS ( 24 runs) | Q5_1  3263.5 GFLOPS ( 24 runs) | Q8_0  3353.4 GFLOPS ( 25 runs)
4096 x 4096: F16   3287.8 GFLOPS ( 24 runs) | F32   3471.8 GFLOPS ( 26 runs)

Running benchmark for all models
This can take a while!

| CPU | OS | Config | Model | Th | Load | Enc. | Commit |
| --- | -- | ------ | ----- | -- | ---- | ---- | ------ |
| <todo> | <todo> |  NEON BLAS | base | 4 | 58 | 293 | 2f52783 |

**Comment by dereklll** [comment]
@[bobqianic](https://github.com/bobqianic)
Can you still have this problem?
Did it（whisper : fix bench regression #1275） really solve the problem? I pull the latest code and still get the same problem.

**Comment by bobqianic** [comment]
> @[bobqianic](https://github.com/bobqianic)
> Do you still have this problem?
> Did it（whisper : fix bench regression #1275） really solve the problem? I pull the latest code and still get the same problem.

It only fixes the issue in bench.cpp. You might need to start a separate issue for this.
